genie-web/src/integTest/java/com/netflix/genie/web/apis/rest/v3/controllers/JobRestControllerIntegrationTest.java

```diff
@@ -726,7 +721,7 @@ private void checkJobExecution(final int documentationId, final String id) {
         if (this.agentExecution) {
             validatableResponse
                 .body(HOST_NAME_PATH, Matchers.notNullValue())
-                .body(PROCESS_ID_PATH, Matchers.notNullValue())
```

Comment:
The agent does send its PID as part of `ReserveJob` or `ClaimJob`, at least according to the protocol. I'm gonna dig a bit and see why they're not coming back here.

Reply:
There's a bug somewhere. When transitioning to `CLAIMED` the agent sends its PID and it is persisted, but later it gets overwritten to `NULL` by some other persistence call. Gonna keep digging..

Reply:
@tgianos mystery solved.. took me a minute because I wasn't really expecting this :-) 

Snippet from `JobEntity.java`:
```java

    @Basic
    @Column(name = "agent_pid")
    @Min(0)
    private Integer agentPid;

    @Basic
    @Column(name = "process_id")
    private Integer processId;
```

There are 2 entity fields (and table columns) to store PID!

`processId` is set by the V3 codepath, and then exposed in the `JobExecution` DTO.
`agent_pid` is set by the V4 codepath, but it's not exposed through any DTO or API.

The new column was introduced here: https://github.com/Netflix/genie/commit/88cfe13

Do you remember what's the rationale for adding a separate column?

We should either expose it through the 'JobExecution` DTO or stop using it and keep the original `processId`.